### PR TITLE
adding youtube adapter

### PR DESCRIPTION
**Pull request by thornycrackers**
Here is the initial work with the adapter with some brief docs on how to use it. I will integrate with the rest soon, just wanted to get this in.

### CODE REVIEW

File: busy_beaver/adapters/youtube.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Where to get your channel id?
+
+Login to youtube and go to https://www.youtube.com/account_advanced, your
+channel id will be displayed.
+
+How to create an api key?
+
+Go to https://console.developers.google.com and create a project. After you can
+visit https://developers.google.com/youtube/registering_an_application#Create_API_Keys
```
[the remark this answers is not in the document]
Comment (uncojohnco) [reply busy_beaver/adapters/youtube.py]: I followed this video and got me up and running
https://www.youtube.com/watch?v=-QMg39gK624

File: busy_beaver/adapters/youtube.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+"""Youtube Adapter.
+
+Docs (TODO: Move into documentation)
```
[the remark this answers is not in the document]
Comment (alysivji) [reply busy_beaver/adapters/youtube.py]: One section per adapter type in the folder's `README.md` makes sense to me.

Clean modules FTW

File: busy_beaver/adapters/youtube.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+api_key = "..."
+channel_id = "..."
+youtube = YoutubeAdapter(api_key=api_key)
+data = youtube.get_latest_videos_from_channel(channel_id)
+videos = data["items"]
+"""
+from typing import Dict
+
+import requests
```
Comment (alysivji) [inline busy_beaver/adapters/youtube.py]: I created a layer of abstraction around the request library for JSON APIs. Utilizing this adapter implements sessions under the hood and returns a `Response` namedtuple which makes life easy. The GitHubAdapter uses this pattern.

My next PR includes a beefed up `internal Requests client` with a `post` method and ability to pass in default headers when creating the client. Should be out within a few hours.

https://github.com/alysivji/busy-beaver/blob/master/busy_beaver/adapters/requests_client.py

File: busy_beaver/adapters/youtube.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+from IPython import embed
+
+
+class YoutubeAdapter:
+    """Small adapter class to make communication with youtube api."""
+
+    def __init__(self, *, api_key: str) -> None:
+        """Initialize with an API Key."""
```
Comment (alysivji) [inline busy_beaver/adapters/youtube.py]: I think comments should only be used if we can't properly express ourselves with code. Which is why I love Python; everything is easy to read.

I think most of these docstrings can be removed as they just repeat the code.

File: busy_beaver/adapters/youtube.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+api_key = "..."
+channel_id = "..."
+youtube = YoutubeAdapter(api_key=api_key)
+data = youtube.get_latest_videos_from_channel(channel_id)
+videos = data["items"]
+"""
+from typing import Dict
+
+import requests
+from IPython import embed
```
[the remark this answers is not in the document]
Comment (thornycrackers) [reply busy_beaver/adapters/youtube.py]: This was leftover from testing, good catch.

**Comment by thornycrackers** [comment]
Updated from the comments, lemme know what you think!

**Comment by alysivji** [comment]
This can be a good place to use vcr.py to write some tests! I have set up a fixture in `conftest.py` that hides API keys passed in thru the header. We will need to modify for this adapter to hide keys passed in thru the body

**Pull request merged**